src/goto-symex/symex_main.cpp

```diff
@@ -67,17 +85,58 @@ void symex_transition(
     // 1. the transition from state.source.pc to "to" is not a backwards goto
     // or
     // 2. we are arriving from an outer loop
+
+    // TODO: This should all be replaced by natural loop analysis.
```

Comment:
Mention the current loop analysis is imprecise when saying it should be replaced.